java/maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java

```diff
@@ -162,15 +166,18 @@ public Boolean getValue() {
             public void setValue(Boolean v) {
                 handle.removePOMModification(operation);
                 modifiedValue = null;
-                String value = v != null ? (v ? "all" : "none") : "all";
+                boolean cosEnabled = v != null ? v : getDefaultValue();
+                String value = cosEnabled ? "all" : "none";
```

Comment:
if we set this to:
`String value = cosEnabled ? "all" : null;`
it would remove the property entirely (and the file too) if it is turned off.

CheckBoxUpdater isn't setting the bold/plain setting properly, but this is a problem for all check boxes in this panel.

Reply:
I'd be inclined to consider that in the NB19 timeframe rather than here. Previous releases have always stored the value, whichever it is. Let's keep that behaviour for now, and just keep the changed default - useful for people switching between 17 & 18, and just in case there's any other edge cases in there.

The bold is kind of correct, if unintuitive - it seems to be whenever the value is set in `nb-configuration` no matter if it matches the default or not.  Another example of this being ternary logic rather than boolean.